sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -125,6 +125,13 @@ abstract class StringRegexExpression extends BinaryExpression
 case class Like(left: Expression, right: Expression, escapeChar: Char)
   extends StringRegexExpression {
 
```

Comment:
For support Column, this construction method has to be added.

Reply:
I think we can use `escapeChar.eval()` in the functions side.

Reply:
Because `escapeChar` is a feature provided myself. I know some background.
The escape char must be a constant. So we didn't support as expression.
I agree the comment wrote by @zhengruifeng . Please check `escapeChar` in the functions API.